lib/puma/server.rb

```diff
@@ -896,11 +900,16 @@ def notify_safely(message)
       begin
         @notify << message
       rescue IOError
+        Thread.current.purge_interrupt_queue if Thread.current.respond_to? :purge_interrupt_queue
         # The server, in another thread, is shutting down
       rescue RuntimeError => e
         # The server, in another thread, has been shut down during the system call
         # https://github.com/puma/puma/pull/1206
-        raise e unless e.message.include?('IOError')
+        if e.message.include?('IOError')
```

Comment:
Got ya :) I just feel it'll be a bit uglier with no real purpose, actually. The `<<` method of IO throws only IOErrors of the StandardError by it's API, so just the`rescue RuntimeError` would be more than enough here. What we have there with `e.message.include?('IOError')` is already over-cautious and safe.